### ISSUE TITLE
Self node in addrbook.json and node tries to dial itself

### ISSUE THREAD

**Issue by re7eal** [tendermint/tendermint#3716]
**Tendermint version** (use `tendermint version` or `git rev-parse --verify HEAD` if installed from source): `v0.31.7`

**ABCI app** (name for built-in, URL for self-written if it's publicly available):
https://github.com/ndidplatform/smart-contract
It should also be reproducible with built-in or any ABCI apps.

**Environment**:
- **OS** (e.g. from /etc/os-release): Linux (Debian 9)
- **Install tools**:
- **Others**: TM in Docker container

**What happened**:
There’s a self node entry in `addrbook.json` and it tries to dial itself and fails.

**What you expected to happen**:
There shouldn't be an entry with self node ID in `addrbook.json` and it should not dial itself.

**Have you tried the latest version**: yes

**How to reproduce it** (as minimally and precisely as possible):
Run multiple nodes with PEX enabled.

**Logs (paste a small part showing an error (< 10 lines) or link a pastebin, gist, etc. containing more of the log file)**:
```
E[2019-06-04|06:01:19.260] dialing failed (attempts: 1): auth failure: secret conn failed: read tcp 172.20.0.5:38760->13.76.0.136:25000: read: connection reset by peer module=pex addr=f0f258b8b0dc09e0dde852600c1f13b4f65fe832@13.76.0.136:25000
E[2019-06-04|06:01:49.233] dialing failed (attempts: 2): auth failure: secret conn failed: EOF module=pex addr=f0f258b8b0dc09e0dde852600c1f13b4f65fe832@13.76.0.136:25000
E[2019-06-04|06:02:19.233] dialing failed (attempts: 3): auth failure: secret conn failed: EOF module=pex addr=f0f258b8b0dc09e0dde852600c1f13b4f65fe832@13.76.0.136:25000
E[2019-06-04|06:02:49.233] dialing failed (attempts: 4): auth failure: secret conn failed: read tcp 172.20.0.5:45554->13.76.0.136:25000: read: connection reset by peer module=pex addr=f0f258b8b0dc09e0dde852600c1f13b4f65fe832@13.76.0.136:25000
```

**Config (you can paste only the changes you've made)**:
```
external_address = "13.76.0.136:25000"

seeds = "a13b8622993d328e6020256113e6386726c95eb3@52.163.184.249:25000"
```

**node command runtime flags**:

**`/dump_consensus_state` output for consensus bugs**

**Anything else we need to know**:

`addrbook.json`
```
{
	"key": "a188b4358bc4c60abaac6bf3",
	"addrs": [
		{
			"addr": {
				"id": "f0f258b8b0dc09e0dde852600c1f13b4f65fe832",
				"ip": "13.76.0.136",
				"port": 25000
			},
			"src": {
				"id": "a13b8622993d328e6020256113e6386726c95eb3",
				"ip": "52.163.184.249",
				"port": 25000
			},
			"attempts": 16,
			"last_attempt": "2019-06-05T00:19:19.233647509Z",
			"last_success": "0001-01-01T00:00:00Z",
			"bucket_type": 1,
			"buckets": [
				225,
				142,
				76
			]
		},
		{
			"addr": {
				"id": "82dac11a9b44a5a2eb1e3b2537f4c93e2e81fd49",
				"ip": "13.67.111.68",
				"port": 25000
			},
			"src": {
				"id": "a13b8622993d328e6020256113e6386726c95eb3",
				"ip": "52.163.184.249",
				"port": 25000
			},
			"attempts": 0,
			"last_attempt": "2019-06-04T06:01:19.258596546Z",
			"last_success": "0001-01-01T00:00:00Z",
			"bucket_type": 1,
			"buckets": [
				21,
				246,
				147
			]
		},
		{
			"addr": {
				"id": "a13b8622993d328e6020256113e6386726c95eb3",
				"ip": "52.163.184.249",
				"port": 25000
			},
			"src": {
				"id": "a13b8622993d328e6020256113e6386726c95eb3",
				"ip": "52.163.184.249",
				"port": 25000
			},
			"attempts": 0,
			"last_attempt": "2019-06-04T06:01:33.168053314Z",
			"last_success": "0001-01-01T00:00:00Z",
			"bucket_type": 1,
			"buckets": [
				152,
				15,
				4
			]
		},
		{
			"addr": {
				"id": "c8f4375561304d482e0226d4d244ba74f01fc042",
				"ip": "13.67.91.196",
				"port": 25000
			},
			"src": {
				"id": "c8f4375561304d482e0226d4d244ba74f01fc042",
				"ip": "13.67.91.196",
				"port": 25000
			},
			"attempts": 0,
			"last_attempt": "2019-06-04T06:01:38.720382132Z",
			"last_success": "0001-01-01T00:00:00Z",
			"bucket_type": 1,
			"buckets": [
				52,
				70,
				182
			]
		}
	]
}
```

`/status`
```
{
  "jsonrpc": "2.0",
  "id": "",
  "result": {
    "node_info": {
      "protocol_version": {
        "p2p": "7",
        "block": "10",
        "app": "2"
      },
      "id": "f0f258b8b0dc09e0dde852600c1f13b4f65fe832",
      "listen_addr": "13.76.0.136:25000",
      "network": "test-chain-7eOWeY",
      "version": "0.31.7",
      "channels": "4020212223303800",
      "moniker": "b26a75d45568",
      "other": {
        "tx_index": "on",
        "rpc_address": "tcp://0.0.0.0:26000"
      }
    },
    "sync_info": {
      "latest_block_hash": "23EEA1FE2F79FA7E684F8B483F7D630B564627DCFFBC39447586222454C02A1E",
      "latest_app_hash": "90D0707E473FD2A4DDC640F11C94C33912BA8140B23FCE6A635766E33E9455E1",
      "latest_block_height": "126642",
      "latest_block_time": "2019-06-05T17:12:17.857901031Z",
      "catching_up": false
    },
    "validator_info": {
      "address": "8970757D071ABDB0B1F0E276234340E65798A636",
      "pub_key": {
        "type": "tendermint/PubKeyEd25519",
        "value": "aCmp1FTl1QPiugyjcLcRUpG7oH9qFrDgpKyFjUVQ1TU="
      },
      "voting_power": "10"
    }
  }
}
```

**Comment by melekes** [comment]
https://github.com/tendermint/tendermint/blob/8b7ca8fd99615cda2f91c385428fb671943888f9/node/node.go#L450 should prevent a node from adding itself to the addrbook. Not sure why it does not work.

We should also consider saving our addresses and private IDs in `addrbook.json`:

```
	ourAddrs   map[string]struct{}
	privateIDs map[p2p.ID]struct{}
```

Right now we only store addresses in `addrbook.json`.

**Comment by re7eal** [comment]
Forgot to mention that `v0.31.5` and below doesn't have this issue though.

**Comment by melekes** [comment]
Can you try `v0.32.0-dev1` (or https://github.com/tendermint/tendermint/pull/3740) and see if it fixes the issue?

**Comment by re7eal** [comment]
@melekes Confirmed that it is fixed in `v0.32.0-dev1`. Thanks for a quick fix!

**Comment by gchaincl** [comment]
@melekes I'm getting a similar error:
```
E[2019-10-29|21:15:37.014] dialing failed (attempts: 7): auth failure: secret conn failed: read tcp 10.156.0.2:47310->10.156.0.2:26656: read: connection reset by peer module=pex addr=9797f5c8aa07d
e58d06124eb037001e63136e4d0@10.156.0.2:26656
```

using `v0.32.4`

**Comment by njmurarka** [comment]
I am seeing this issue still. I erroneously posted a report of this in the COSMOS-SDK issue database, that I have since closed. But most certainly, as of today still, with the following two versions in my go.mod, I am seeing this happen sporadically:

cosmos-sdk v0.38.1
tendermint v0.33.0

Here is the occasional output I get from all four validators on my four validator chain:

```
-------- daemon-3 (3.122.11.179) ----------

E[2020-03-15|21:27:08.497] dialing failed (attempts: 8): auth failure: secret conn failed: read tcp 172.17.0.2:59290->3.122.11.179:26656: read: connection reset by peer module=pex addr=1304d02f68fc77b26b7ec0e136bfbbd63eb26cd2@3.122.11.179:26656

-------- daemon-2 (52.221.96.154) ----------

E[2020-03-15|21:27:08.767] dialing failed (attempts: 8): auth failure: secret conn failed: read tcp 172.17.0.2:44414->52.221.96.154:26656: read: connection reset by peer module=pex addr=14cb1b6d786822407f198b4b59ba40aedfb163d1@52.221.96.154:26656

-------- daemon-1 (13.127.191.110) ----------

E[2020-03-15|21:27:09.175] dialing failed (attempts: 8): auth failure: secret conn failed: EOF module=pex addr=3227619179f9ae569989d303636eb45566ad7544@13.127.191.110:26656

-------- daemon-0 (34.231.181.10) ----------

E[2020-03-15|21:27:11.227] dialing failed (attempts: 8): auth failure: secret conn failed: EOF module=pex addr=74d8eb429cb91e3643f8a6311ecda6776de37b57@34.231.181.10:26656
```

Was this problem resolved? I see that it is stated above that it was, in v0.32.0-dev1. What is the most obvious explanation for why I would be seeing it then?

Here is a link to the original COSMOS-SDK issue:

https://github.com/cosmos/cosmos-sdk/issues/5806

**Comment by njmurarka** [comment]
Just noticed it was re-opened. Are others able to reproduce?

**Comment by jingnanzhou** [comment]
I have the same issue
E[2020-05-17|19:36:37.098] dialing failed (attempts: 1): self ID<5bca43042ee8072fbc7c031141123d44684b1570> module=pex addr=5bca43042ee8072fbc7c031141123d44684b1570@192.168.1.43:26661

tendermint version 0.33.2

**Comment by banishee** [comment]
so did I
```
E[2020-06-17|08:49:38.070][1] dialing failed (attempts: 1): auth failure: secret conn failed: EOF module=pex addr=b7c6bdfe0c3a6c1c68d6d6849f1b60f566e189dd@192.168.13.120:26656
E[2020-06-17|08:50:38.273][1] dialing failed (attempts: 2): auth failure: secret conn failed: EOF module=pex addr=b7c6bdfe0c3a6c1c68d6d6849f1b60f566e189dd@192.168.13.120:26656
E[2020-06-17|08:51:08.069][1] dialing failed (attempts: 3): auth failure: secret conn failed: EOF module=pex addr=b7c6bdfe0c3a6c1c68d6d6849f1b60f566e189dd@192.168.13.120:26656
E[2020-06-17|08:51:38.069][1] dialing failed (attempts: 4): auth failure: secret conn failed: EOF module=pex addr=b7c6bdfe0c3a6c1c68d6d6849f1b60f566e189dd@192.168.13.120:26656
E[2020-06-17|08:52:08.069][1] dialing failed (attempts: 5): auth failure: secret conn failed: EOF module=pex addr=b7c6bdfe0c3a6c1c68d6d6849f1b60f566e189dd@192.168.13.120:26656
E[2020-06-17|08:53:38.069][1] dialing failed (attempts: 6): auth failure: secret conn failed: read tcp 172.18.0.5:33102->192.168.13.120:26656: read: connection reset by peer module=pex addr=b7c6bdfe0c3a6c1c68d6d6849f1b60f566e189dd@192.168.13.120:26656
E[2020-06-17|08:55:08.070][1] dialing failed (attempts: 7): auth failure: secret conn failed: EOF module=pex addr=b7c6bdfe0c3a6c1c68d6d6849f1b60f566e189dd@192.168.13.120:26656
```

**Comment by pkrasam** [comment]
I am running into similar issues -- does anyone have a workaround?

**Comment by fedekunze** [comment]
we are running into the same issue on Ethermint. Tendermint version `0.33.6` cc: @marbar3778 @tessr

**Comment by ColinSchwarz** [comment]
@marbar3778 @tessr thanks for bring this up! We are in direct communications with the core Tendermint team about this and will look into fixing it in the next few days.

**Comment by araskachoi** [comment]
> we are running into the same issue on Ethermint. Tendermint version `0.33.6` cc: @marbar3778 @tessr
> 

In addition to @fedekunze comment, the addrbooks are as follows for the nodes:

```
node0: # cat addrbook.json 
{
        "key": "eb5f57e4cd6ae932756cb836",
        "addrs": [
                {
                        "addr": {
                                "id": "156fe8022faa1923a40c216ee4281e0a7da7134e",
                                "ip": "159.203.42.139",
                                "port": 26656
                        },
                        "src": {
                                "id": "49f8fc19689762779251d58a83d9bdc776727283",
                                "ip": "0.0.0.0",
                                "port": 26656
                        },
                        "buckets": [
                                28
                        ],
                        "attempts": 29,
                        "bucket_type": 1,
                        "last_attempt": "2020-08-04T16:28:52.962294296Z",
                        "last_success": "0001-01-01T00:00:00Z",
                        "last_ban_time": "0001-01-01T00:00:00Z"
                },
                {
                        "addr": {
                                "id": "49f8fc19689762779251d58a83d9bdc776727283",
                                "ip": "165.227.35.15",
                                "port": 26656
                        },
                        "src": {
                                "id": "156fe8022faa1923a40c216ee4281e0a7da7134e",
                                "ip": "0.0.0.0",
                                "port": 26656
                        },
                        "buckets": [
                                14
                        ],
                        "attempts": 11,
                        "bucket_type": 1,
                        "last_attempt": "2020-08-03T21:48:58.350239574Z",
                        "last_success": "0001-01-01T00:00:00Z",
                        "last_ban_time": "0001-01-01T00:00:00Z"
                },
                {
                        "addr": {
                                "id": "743833c99687854c4d86668f25ac7c1c90233ab9",
                                "ip": "178.128.233.8",
                                "port": 26656
                        },
                        "src": {
                                "id": "156fe8022faa1923a40c216ee4281e0a7da7134e",
                                "ip": "0.0.0.0",
                                "port": 26656
                        },
                        "buckets": [
                                191
                        ],
                        "attempts": 11,
                        "bucket_type": 1,
                        "last_attempt": "2020-08-03T21:48:58.350363237Z",
                        "last_success": "0001-01-01T00:00:00Z",
                        "last_ban_time": "0001-01-01T00:00:00Z"
                }
        ]
}
```

```
node1: # cat addrbook.json 
{
        "key": "f53d56642c7903a2f0b17c00",
        "addrs": [
                {
                        "addr": {
                                "id": "156fe8022faa1923a40c216ee4281e0a7da7134e",
                                "ip": "159.203.42.139",
                                "port": 26656
                        },
                        "src": {
                                "id": "743833c99687854c4d86668f25ac7c1c90233ab9",
                                "ip": "0.0.0.0",
                                "port": 26656
                        },
                        "buckets": [
                                228
                        ],
                        "attempts": 0,
                        "bucket_type": 1,
                        "last_attempt": "2020-08-03T22:10:32.304517921Z",
                        "last_success": "0001-01-01T00:00:00Z",
                        "last_ban_time": "0001-01-01T00:00:00Z"
                },
                {
                        "addr": {
                                "id": "49f8fc19689762779251d58a83d9bdc776727283",
                                "ip": "165.227.35.15",
                                "port": 26656
                        },
                        "src": {
                                "id": "743833c99687854c4d86668f25ac7c1c90233ab9",
                                "ip": "0.0.0.0",
                                "port": 26656
                        },
                        "buckets": [
                                100
                        ],
                        "attempts": 1,
                        "bucket_type": 1,
                        "last_attempt": "2020-08-03T22:10:32.363432292Z",
                        "last_success": "0001-01-01T00:00:00Z",
                        "last_ban_time": "0001-01-01T00:00:00Z"
                },
                {
                        "addr": {
                                "id": "743833c99687854c4d86668f25ac7c1c90233ab9",
                                "ip": "178.128.233.8",
                                "port": 26656
                        },
                        "src": {
                                "id": "156fe8022faa1923a40c216ee4281e0a7da7134e",
                                "ip": "0.0.0.0",
                                "port": 26656
                        },
                        "buckets": [
                                194
                        ],
                        "attempts": 16,
                        "bucket_type": 1,
                        "last_attempt": "2020-08-04T16:29:02.34479974Z",
                        "last_success": "0001-01-01T00:00:00Z",
                        "last_ban_time": "0001-01-01T00:00:00Z"
                }
        ]
}
```

```
node2: # cat addrbook.json 
{
        "key": "4afe232c2d7bdd34afb3ce94",
        "addrs": [
                {
                        "addr": {
                                "id": "156fe8022faa1923a40c216ee4281e0a7da7134e",
                                "ip": "159.203.42.139",
                                "port": 26656
                        },
                        "src": {
                                "id": "49f8fc19689762779251d58a83d9bdc776727283",
                                "ip": "0.0.0.0",
                                "port": 26656
                        },
                        "buckets": [
                                19
                        ],
                        "attempts": 0,
                        "bucket_type": 1,
                        "last_attempt": "2020-08-03T22:10:32.780030702Z",
                        "last_success": "0001-01-01T00:00:00Z",
                        "last_ban_time": "0001-01-01T00:00:00Z"
                },
                {
                        "addr": {
                                "id": "743833c99687854c4d86668f25ac7c1c90233ab9",
                                "ip": "178.128.233.8",
                                "port": 26656
                        },
                        "src": {
                                "id": "49f8fc19689762779251d58a83d9bdc776727283",
                                "ip": "0.0.0.0",
                                "port": 26656
                        },
                        "buckets": [
                                36
                        ],
                        "attempts": 0,
                        "bucket_type": 1,
                        "last_attempt": "2020-08-03T22:10:32.780052601Z",
                        "last_success": "0001-01-01T00:00:00Z",
                        "last_ban_time": "0001-01-01T00:00:00Z"
                },
                {
                        "addr": {
                                "id": "49f8fc19689762779251d58a83d9bdc776727283",
                                "ip": "165.227.35.15",
                                "port": 26656
                        },
                        "src": {
                                "id": "743833c99687854c4d86668f25ac7c1c90233ab9",
                                "ip": "0.0.0.0",
                                "port": 26656
                        },
                        "buckets": [
                                173
                        ],
                        "attempts": 16,
                        "bucket_type": 1,
                        "last_attempt": "2020-08-04T16:29:02.786251548Z",
                        "last_success": "0001-01-01T00:00:00Z",
                        "last_ban_time": "0001-01-01T00:00:00Z"
                }
        ]
}
```

**Comment by erikgrinaker** [comment]
I believe this is caused by the P2P address handling being fundamentally broken, and e.g. conflating the bind (listen) address with the advertised address and the external address. I am not sure if we can fix this without rewriting parts of the P2P subsystem, but we may be able to come up with a workaround. However, I believe the error is harmless, and thus not a priority at the moment - if someone is experiencing actual node failures because of this then please let us know.

**Comment by xla** [comment]
Ref #2067 for tracking purposes.

**Comment by sunnyk56** [comment]
I have the same issue

``2:12PM ERR dialing failed (attempts: 1): auth failure: secret conn failed: read tcp 147.182.190.16:53388->137.184.192.172:26656: read: connection reset by peer addr={"id":"ba9168215dc644065a7a5fface0c976582c7e529","ip":"137.184.192.172","port":26656} module=pex
2:12PM ERR dialing failed (attempts: 1): dial tcp 78.31.65.76:26656: connect: connection refused addr={"id":"b1e6561ba7c345c8b7cb4209116c1fcac548d2e0","ip":"78.31.65.76","port":26656} module=pex
2:12PM ERR dialing failed (attempts: 1): dial tcp 194.163.168.71:26656: connect: connection refused addr={"id":"f17dbfcaa66822bf48140b952d1f93d708d37fee","ip":"194.163.168.71","port":26656} module=pex
2:12PM ERR dialing failed (attempts: 1): dial tcp 23.88.33.187:26656: connect: connection refused addr={"id":"6a4ff5ac3d092ab375e9d2a2df6f128592956d62","ip":"23.88.33.187","port":26656} module=pex
2:12PM ERR dialing failed (attempts: 1): dial tcp 161.97.127.66:26656: connect: connection refused addr={"id":"94f6697f0f98f15a09f4d8e40317bf10435fffea","ip":"161.97.127.66","port":26656} module=pex
2:12PM ERR dialing failed (attempts: 1): dial tcp 144.91.80.254:26656: connect: connection refused addr={"id":"5bb52386d004a89d0adeb185a9db1d8b93812504","ip":"144.91.80.254","port":26656} module=pex
2:12PM ERR dialing failed (attempts: 1): dial tcp 185.182.9.228:26656: connect: connection refused addr={"id":"7b50e55759a52651d4f9932463778fc66c37fc0e","ip":"185.182.9.228","port":26656} module=pex
2:12PM ERR dialing failed (attempts: 1): dial tcp 65.21.239.101:26656: connect: connection refused addr={"id":"894138d9b9b0d0c6d9ab91146ed537751f504383","ip":"65.21.239.101","port":26656} module=pex
2:12PM ERR dialing failed (attempts: 1): dial tcp 95.111.239.28:26656: connect: connection refused addr={"id":"b755daaf87b4ac116766e0991314799ec98fa76e","ip":"95.111.239.28","port":26656} module=pex
2:12PM ERR dialing failed (attempts: 1): dial tcp 161.97.98.155:26656: connect: connection refused addr={"id":"7fba7be279509d855e69e4acc185c2f218264e07","ip":"161.97.98.155","port":26656} module=pex
2:12PM ERR dialing failed (attempts: 1): auth failure: conn.ID (999e7247a10b143a5badbec12f9b47c60628ee6d) dialed ID (2efd13820d3c2d1e145c51d2bc3f7b7cbdbf80e3) mismatch addr={"id":"2efd13820d3c2d1e145c51d2bc3f7b7cbdbf80e3","ip":"45.63.115.66","port":26656} module=pex
2:12PM ERR dialing failed (attempts: 1): auth failure: secret conn failed: read tcp 147.182.190.16:39988->167.86.115.15:26656: read: connection reset by peer addr={"id":"f1d5308513b0f2cac0ef8c6322d6e74f1bb2c5ee","ip":"167.86.115.15","port":26656} module=pex
2:12PM ERR dialing failed (attempts: 1): auth failure: secret conn failed: read tcp 147.182.190.16:45426->178.124.214.192:26656: read: connection reset by peer addr={"id":"9900ae6425a3af67e92989458f144d6da07ba31f","ip":"178.124.214.192","port":26656} module=pex
2:12PM ERR dialing failed (attempts: 1): dial tcp 49.12.224.94:26656: i/o timeout addr={"id":"2622fd3dd274d65ddd2e81b8e4a7edcbe94778c9","ip":"49.12.224.94","port":26656} module=pex
2:12PM ERR failed signing vote err="error signing vote: round regression at height 493529. Got 0, last round 10" height=493529 module=consensus round=0 vote={"block_id":{"hash":"","parts":{"hash":"","total":0}},"height":493529,"round":0,"signature":null,"timestamp":"2021-11-02T14:12:09.174876648Z","type":1,"validator_address":"A58CF2FF29FF6CF89C3684A6FF327F2EA0F9FF2D","validator_index":0}
2:12PM ERR failed signing vote err="error signing vote: round regression at height 493529. Got 0, last round 10" height=493529 module=consensus round=0 vote={"block_id":{"hash":"","parts":{"hash":"","total":0}},"height":493529,"round":0,"signature":null,"timestamp":"2021-11-02T14:12:09.274098408Z","type":2,"validator_address":"A58CF2FF29FF6CF89C3684A6FF327F2EA0F9FF2D","validator_index":0}
2:12PM ERR failed signing vote err="error signing vote: round regression at height 493529. Got 1, last round 10" height=493529 module=consensus round=1 vote={"block_id":{"hash":"","parts":{"hash":"","total":0}},"height":493529,"round":1,"signature":null,"timestamp":"2021-11-02T14:12:09.353382007Z","type":1,"validator_address":"A58CF2FF29FF6CF89C3684A6FF327F2EA0F9FF2D","validator_index":0}
2:12PM ERR failed signing vote err="error signing vote: round regression at height 493529. Got 1, last round 10" height=493529 module=consensus round=1 vote={"block_id":{"hash":"","parts":{"hash":"","total":0}},"height":493529,"round":1,"signature":null,"timestamp":"2021-11-02T14:12:09.488675311Z","type":2,"validator_address":"A58CF2FF29FF6CF89C3684A6FF327F2EA0F9FF2D","validator_index":0}
2:12PM ERR failed signing vote err="error signing vote: round regression at height 493529. Got 2, last round 10" height=493529 module=consensus round=2 vote={"block_id":{"hash":"","parts":{"hash":"","total":0}},"height":493529,"round":2,"signature":null,"timestamp":"2021-11-02T14:12:09.624045721Z","type":1,"validator_address":"A58CF2FF29FF6CF89C3684A6FF327F2EA0F9FF2D","validator_index":0}
2:12PM ERR failed signing vote err="error signing vote: round regression at height 493529. Got 2, last round 10" height=493529 module=consensus round=2 vote={"block_id":{"hash":"","parts":{"hash":"","total":0}},"height":493529,"round":2,"signature":null,"timestamp":"2021-11-02T14:12:09.772410304Z","type":2,"validator_address":"A58CF2FF29FF6CF89C3684A6FF327F2EA0F9FF2D","validator_index":0}
2:12PM ERR failed signing vote err="error signing vote: round regression at height 493529. Got 3, last round 10" height=493529 module=consensus round=3 vote={"block_id":{"hash":"","parts":{"hash":"","total":0}},"height":493529,"round":3,"signature":null,"timestamp":"2021-11-02T14:12:09.962546323Z","type":1,"validator_address":"A58CF2FF29FF6CF89C3684A6FF327F2EA0F9FF2D","validator_index":0}
2:12PM ERR failed signing vote err="error signing vote: round regression at height 493529. Got 3, last round 10" height=493529 module=consensus round=3 vote={"block_id":{"hash":"","parts":{"hash":"","total":0}},"height":493529,"round":3,"signature":null,"timestamp":"2021-11-02T14:12:10.099228288Z","type":2,"validator_address":"A58CF2FF29FF6CF89C3684A6FF327F2EA0F9FF2D","validator_index":0}
2:12PM ERR mempool is full: number of txs 5000 (max: 5000), total txs bytes 4609252 (max: 1073741824) module=mempool
2:12PM ERR mempool is full: number of txs 5000 (max: 5000), total txs bytes 4609252 (max: 1073741824) module=mempool
2:12PM ERR mempool is full: number of txs 5000 (max: 5000), total txs bytes 4609252 (max: 1073741824) module=mempool
2:12PM ERR mempool is full: number of txs 5000 (max: 5000), total txs bytes 4609252 (max: 1073741824) module=mempool
2:12PM ERR mempool is full: number of txs 5000 (max: 5000), total txs bytes 4609252 (max: 1073741824) module=mempool
2:12PM ERR failed signing vote err="error signing vote: round regression at height 493529. Got 4, last round 10" height=493529 module=consensus round=4 vote={"block_id":{"hash":"","parts":{"hash":"","total":0}},"height":493529,"round":4,"signature":null,"timestamp":"2021-11-02T14:12:10.390211016Z","type":1,"validator_address":"A58CF2FF29FF6CF89C3684A6FF327F2EA0F9FF2D","validator_index":0}
2:12PM ERR failed signing vote err="error signing vote: round regression at height 493529. Got 4, last round 10" height=493529 module=consensus round=4 vote={"block_id":{"hash":"","parts":{"hash":"","total":0}},"height":493529,"round":4,"signature":null,"timestamp":"2021-11-02T14:12:10.538861281Z","type":2,"validator_address":"A58CF2FF29FF6CF89C3684A6FF327F2EA0F9FF2D","validator_index":0}
2:12PM ERR dialing failed (attempts: 1): dial tcp 185.239.208.91:26656: connect: connection refused addr={"id":"9a2ebbaea6183895d431462583b0b25ef8e5ccdf","ip":"185.239.208.91","port":26656} module=pex
2:12PM ERR dialing failed (attempts: 1): dial tcp 185.225.232.215:26656: connect: connection refused addr={"id":"6d5e35fa07a276fb50b8a8f9e5ea697530c6c5e7","ip":"185.225.232.215","port":26656} module=pex
2:12PM ERR dialing failed (attempts: 1): dial tcp 185.218.126.215:26656: connect: connection refused addr={"id":"c765406da638a7b53a409407a71c6c1019625d96","ip":"185.218.126.215","port":26656} module=pex
2:12PM ERR dialing failed (attempts: 2): dial tcp 144.91.80.254:26656: connect: connection refused addr={"id":"5bb52386d004a89d0adeb185a9db1d8b93812504","ip":"144.91.80.254","port":26656} module=pex
2:12PM ERR dialing failed (attempts: 1): dial tcp 95.217.223.153:26656: connect: connection refused addr={"id":"b589bfc5f2b454e2d982193a9466adcfca0713c6","ip":"95.217.223.153","port":26656} module=pex
2:12PM ERR dialing failed (attempts: 1): dial tcp 167.86.100.178:26656: connect: connection refused addr={"id":"197fb9a4172341984fc75e6ad92e0f7e584e9543","ip":"167.86.100.178","port":26656} module=pex
2:12PM ERR dialing failed (attempts: 2): auth failure: conn.ID (999e7247a10b143a5badbec12f9b47c60628ee6d) dialed ID (dcb8baf669c159b70ad59d84a193f52a422e1212) mismatch addr={"id":"dcb8baf669c159b70ad59d84a193f52a422e1212","ip":"45.63.115.66","port":26656} module=pex
2:12PM ERR dialing failed (attempts: 1): auth failure: conn.ID (0571fb99e34bfebf9d3b4894436d65d60fe9b7f7) dialed ID (1a6a6d19353a3439fd5b2619c12372cee535e380) mismatch addr={"id":"1a6a6d19353a3439fd5b2619c12372cee535e380","ip":"185.239.209.9","port":26656} module=pex
2:12PM ERR dialing failed (attempts: 1): auth failure: conn.ID (3ca85c13ebe5d4d341ff2f8139685085e217d886) dialed ID (7266e6b5c44a2cd20af912d084605d804dd953b7) mismatch addr={"id":"7266e6b5c44a2cd20af912d084605d804dd953b7","ip":"65.108.55.169","port":26656} module=pex
2:12PM ERR dialing failed (attempts: 1): dial tcp 70.34.200.242:26656: i/o timeout addr={"id":"20ac05272a47b767aade9a71eda9e305db12656b","ip":"70.34.200.242","port":26656} module=pex
``

cosmos-sdk v0.42.1
tendermint v0.34.8

**Comment by alexeysuvorov** [comment]
Unfortunately I have the same error happened after one of 4 nodes was disconnected because of network failure. Main nodes constantly add itself to address book and tries to call itself.

`Dec 03 15:48:31 metachain-host-1 cosmovisor[24132]: 3:48PM INF We need more addresses. Sending pexRequest to random peer module=pex peer={"Data":{},"Logger":{}}
Dec 03 15:48:31 metachain-host-1 cosmovisor[24132]: 3:48PM INF Dialing peer address={"id":"d77ab4fbe92f8320c42c437bf67bdc2f31f154d2","ip":"35.202.1.92","port":26656} module=p2p
Dec 03 15:48:31 metachain-host-1 cosmovisor[24132]: 3:48PM INF Inbound Peer rejected err="filtered CONN<35.202.1.92:58920>: duplicate CONN<35.202.1.92:58920>" module=p2p numPeers=1
Dec 03 15:48:31 metachain-host-1 cosmovisor[24132]: 3:48PM ERR dialing failed (attempts: 1): auth failure: secret conn failed: EOF addr={"id":"d77ab4fbe92f8320c42c437bf67bdc2f31f154d2","ip":"35.202.1.92","port":26656} module=pex
Dec 03 15:49:01 metachain-host-1 cosmovisor[24132]: 3:49PM INF Ensure peers module=pex numDialing=0 numInPeers=1 numOutPeers=0 numToDial=10
Dec 03 15:49:01 metachain-host-1 cosmovisor[24132]: 3:49PM INF Will dial address addr={"id":"d77ab4fbe92f8320c42c437bf67bdc2f31f154d2","ip":"35.202.1.92","port":26656} module=pex`

**Comment by effofxprime** [comment]
I believe that I am facing the same error.  I am not listed the persistent pers list and I am not seeing dialing errors or rejected peer errors.   /status is below, but my ID is "fa1144b7a13a93660206d30f4f9e45d3dffe5087".  Is there something I can attempt to try to see if my addrbook can be populated correctly?

```
cat config/addrbook.json
{
        "key": "465677c1b2c683a4b5b9c947",
        "addrs": [
                {
                        "addr": {
                                "id": "d45e9dd8878d7c22d59ded3557f61da37420a4c6",
                                "ip": "95.217.118.211",
                                "port": 26656
                        },
                        "src": {
                                "id": "fa1144b7a13a93660206d30f4f9e45d3dffe5087",
                                "ip": "0.0.0.0",
                                "port": 26656
                        },
                        "buckets": [
                                111
                        ],
                        "attempts": 7,
                        "bucket_type": 1,
                        "last_attempt": "2021-12-18T09:32:41.606901345Z",
                        "last_success": "0001-01-01T00:00:00Z",
                        "last_ban_time": "0001-01-01T00:00:00Z"
                },
                {
                        "addr": {
                                "id": "cae7d9d21c1752300277eab72d861b0c6638b2e3",
                                "ip": "164.68.119.151",
                                "port": 26656
                        },
                        "src": {
                                "id": "fa1144b7a13a93660206d30f4f9e45d3dffe5087",
                                "ip": "0.0.0.0",
                                "port": 26656
                        },
                        "buckets": [
                                50
                        ],
                        "attempts": 0,
                        "bucket_type": 2,
                        "last_attempt": "2021-12-18T15:53:15.749889087Z",
                        "last_success": "2021-12-18T15:53:15.749889087Z",
                        "last_ban_time": "0001-01-01T00:00:00Z"
                },
                {
                        "addr": {
                                "id": "fa1144b7a13a93660206d30f4f9e45d3dffe5087",
                                "ip": "45.33.55.23",
                                "port": 26656
                        },
                        "src": {
                                "id": "209688f5bccb88f6397a97cc11ab545a014aa559",
                                "ip": "0.0.0.0",
                                "port": 26656
                        },
                        "buckets": [
                                138
                        ],
                        "attempts": 7,
                        "bucket_type": 1,
                        "last_attempt": "2021-12-18T09:31:14.134339918Z",
                        "last_success": "0001-01-01T00:00:00Z",
                        "last_ban_time": "0001-01-01T00:00:00Z"
                },
                {
                        "addr": {
                                "id": "7a44ea6ecb59b0e4bd01b58a75163ec64b164bb4",
                                "ip": "63.210.148.24",
                                "port": 26656
                        },
                        "src": {
                                "id": "209688f5bccb88f6397a97cc11ab545a014aa559",
                                "ip": "0.0.0.0",
                                "port": 26656
                        },
                        "buckets": [
                                19
                        ],
                        "attempts": 0,
                        "bucket_type": 2,
                        "last_attempt": "2021-12-18T19:09:11.672617215Z",
                        "last_success": "2021-12-18T19:09:11.672617215Z",
                        "last_ban_time": "0001-01-01T00:00:00Z"
                },
                {
                        "addr": {
                                "id": "3bf3d98dfd4000dd5ff8189882a9f96848b99b87",
                                "ip": "137.220.60.196",
                                "port": 26656
                        },
                        "src": {
                                "id": "209688f5bccb88f6397a97cc11ab545a014aa559",
                                "ip": "0.0.0.0",
                                "port": 26656
                        },
                        "buckets": [
                                11
                        ],
                        "attempts": 0,
                        "bucket_type": 2,
                        "last_attempt": "2021-12-18T19:29:37.745740391Z",
                        "last_success": "2021-12-18T19:29:37.745740391Z",
                        "last_ban_time": "0001-01-01T00:00:00Z"
                },
                {
                        "addr": {
                                "id": "057fa262fe2030cc6e9095dc52d15b79ffcb923d",
                                "ip": "142.115.20.25",
                                "port": 26656
                        },
                        "src": {
                                "id": "209688f5bccb88f6397a97cc11ab545a014aa559",
                                "ip": "0.0.0.0",
                                "port": 26656
                        },
                        "buckets": [
                                42
                        ],
                        "attempts": 0,
                        "bucket_type": 2,
                        "last_attempt": "2021-12-18T16:57:29.566490769Z",
                        "last_success": "2021-12-18T16:57:29.566490769Z",
                        "last_ban_time": "0001-01-01T00:00:00Z"
                },
                {
                        "addr": {
                                "id": "209688f5bccb88f6397a97cc11ab545a014aa559",
                                "ip": "137.184.92.115",
                                "port": 26656
                        },
                        "src": {
                                "id": "fa1144b7a13a93660206d30f4f9e45d3dffe5087",
                                "ip": "0.0.0.0",
                                "port": 26656
                        },
                        "buckets": [
                                35
                        ],
                        "attempts": 0,
                        "bucket_type": 2,
                        "last_attempt": "2021-12-18T19:06:00.669650479Z",
                        "last_success": "2021-12-18T19:06:00.669650479Z",
                        "last_ban_time": "0001-01-01T00:00:00Z"
                }
        ]
}
```

**/status**
```
vidulumd status
{"NodeInfo":{"protocol_version":{"p2p":"8","block":"11","app":"0"},"id":"fa1144b7a13a93660206d30f4f9e45d3dffe5087","listen_addr":"tcp://0.0.0.0:26656","network":"vidulum-1","version":"0.34.12","channels":"40202122233038606100","moniker":"moneta","other":{"tx_index":"on","rpc_address":"tcp://127.0.0.1:26657"}},"SyncInfo":{"latest_block_hash":"150231A3D5F0A8AB34C7085F9D24747ACA570E4DA2DA81B36B909BCAAF64351A","latest_app_hash":"D1561A350A03013A702FD440ED7B6F9CB5FD8535EA011D28C618A37747892A8F","latest_block_height":"826730","latest_block_time":"2021-12-18T19:43:21.544345923Z","earliest_block_hash":"5614A5F7D398CCAC49EFB255D1F92421891B725808E31421317BDD519D35F7CA","earliest_app_hash":"E3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B934CA495991B7852B855","earliest_block_height":"1","earliest_block_time":"2021-10-23T00:00:00Z","catching_up":false},"ValidatorInfo":{"Address":"4C1438CDA672B766D7BA69A79DC462DD389875BE","PubKey":{"type":"tendermint/PubKeyEd25519","value":"WlrhadzFXUtjvJLUvlx9eypjBMJwImF2divAGDXemzg="},"VotingPower":"13878"}}
```

**config.toml**
```
cat config/config.toml
# This is a TOML config file.
# For more information, see https://github.com/toml-lang/toml

# NOTE: Any path below can be absolute (e.g. "/var/myawesomeapp/data") or
# relative to the home directory (e.g. "data"). The home directory is
# "$HOME/.tendermint" by default, but could be changed via $TMHOME env variable
# or --home cmd flag.

#######################################################################
###                   Main Base Config Options                      ###
#######################################################################

# TCP or UNIX socket address of the ABCI application,
# or the name of an ABCI application compiled in with the Tendermint binary
proxy_app = "tcp://127.0.0.1:26658"

# A custom human readable name for this node
moniker = "Moneta By Erialos"

# If this node is many blocks behind the tip of the chain, FastSync
# allows them to catchup quickly by downloading blocks in parallel
# and verifying their commits
fast_sync = true

# Database backend: goleveldb | cleveldb | boltdb | rocksdb | badgerdb
# * goleveldb (github.com/syndtr/goleveldb - most popular implementation)
#   - pure go
#   - stable
# * cleveldb (uses levigo wrapper)
#   - fast
#   - requires gcc
#   - use cleveldb build tag (go build -tags cleveldb)
# * boltdb (uses etcd's fork of bolt - github.com/etcd-io/bbolt)
#   - EXPERIMENTAL
#   - may be faster is some use-cases (random reads - indexer)
#   - use boltdb build tag (go build -tags boltdb)
# * rocksdb (uses github.com/tecbot/gorocksdb)
#   - EXPERIMENTAL
#   - requires gcc
#   - use rocksdb build tag (go build -tags rocksdb)
# * badgerdb (uses github.com/dgraph-io/badger)
#   - EXPERIMENTAL
#   - use badgerdb build tag (go build -tags badgerdb)
db_backend = "goleveldb"

# Database directory
db_dir = "data"

# Output level for logging, including package level options
log_level = "info"

# Output format: 'plain' (colored text) or 'json'
log_format = "plain"

##### additional base config options #####

# Path to the JSON file containing the initial validator set and other meta data
genesis_file = "config/genesis.json"

# Path to the JSON file containing the private key to use as a validator in the consensus protocol
priv_validator_key_file = "config/priv_validator_key.json"

# Path to the JSON file containing the last sign state of a validator
priv_validator_state_file = "data/priv_validator_state.json"

# TCP or UNIX socket address for Tendermint to listen on for
# connections from an external PrivValidator process
priv_validator_laddr = ""

# Path to the JSON file containing the private key to use for node authentication in the p2p protocol
node_key_file = "config/node_key.json"

# Mechanism to connect to the ABCI application: socket | grpc
abci = "socket"

# If true, query the ABCI app on connecting to a new peer
# so the app can decide if we should keep the connection or not
filter_peers = true


#######################################################################
###                 Advanced Configuration Options                  ###
#######################################################################

#######################################################
###       RPC Server Configuration Options          ###
#######################################################
[rpc]

# TCP or UNIX socket address for the RPC server to listen on
laddr = "tcp://127.0.0.1:26657"

# A list of origins a cross-domain request can be executed from
# Default value '[]' disables cors support
# Use '["*"]' to allow any origin
cors_allowed_origins = []

# A list of methods the client is allowed to use with cross-domain requests
cors_allowed_methods = ["HEAD", "GET", "POST", ]

# A list of non simple headers the client is allowed to use with cross-domain requests
cors_allowed_headers = ["Origin", "Accept", "Content-Type", "X-Requested-With", "X-Server-Time", ]

# TCP or UNIX socket address for the gRPC server to listen on
# NOTE: This server only supports /broadcast_tx_commit
grpc_laddr = ""

# Maximum number of simultaneous connections.
# Does not include RPC (HTTP&WebSocket) connections. See max_open_connections
# If you want to accept a larger number than the default, make sure
# you increase your OS limits.
# 0 - unlimited.
# Should be < {ulimit -Sn} - {MaxNumInboundPeers} - {MaxNumOutboundPeers} - {N of wal, db and other open files}
# 1024 - 40 - 10 - 50 = 924 = ~900
grpc_max_open_connections = 900

# Activate unsafe RPC commands like /dial_seeds and /unsafe_flush_mempool
unsafe = false

# Maximum number of simultaneous connections (including WebSocket).
# Does not include gRPC connections. See grpc_max_open_connections
# If you want to accept a larger number than the default, make sure
# you increase your OS limits.
# 0 - unlimited.
# Should be < {ulimit -Sn} - {MaxNumInboundPeers} - {MaxNumOutboundPeers} - {N of wal, db and other open files}
# 1024 - 40 - 10 - 50 = 924 = ~900
max_open_connections = 900

# Maximum number of unique clientIDs that can /subscribe
# If you're using /broadcast_tx_commit, set to the estimated maximum number
# of broadcast_tx_commit calls per block.
max_subscription_clients = 100

# Maximum number of unique queries a given client can /subscribe to
# If you're using GRPC (or Local RPC client) and /broadcast_tx_commit, set to
# the estimated # maximum number of broadcast_tx_commit calls per block.
max_subscriptions_per_client = 5

# How long to wait for a tx to be committed during /broadcast_tx_commit.
# WARNING: Using a value larger than 10s will result in increasing the
# global HTTP write timeout, which applies to all connections and endpoints.
# See https://github.com/tendermint/tendermint/issues/3435
timeout_broadcast_tx_commit = "10s"

# Maximum size of request body, in bytes
max_body_bytes = 1000000

# Maximum size of request header, in bytes
max_header_bytes = 1048576

# The path to a file containing certificate that is used to create the HTTPS server.
# Might be either absolute path or path related to Tendermint's config directory.
# If the certificate is signed by a certificate authority,
# the certFile should be the concatenation of the server's certificate, any intermediates,
# and the CA's certificate.
# NOTE: both tls_cert_file and tls_key_file must be present for Tendermint to create HTTPS server.
# Otherwise, HTTP server is run.
tls_cert_file = ""

# The path to a file containing matching private key that is used to create the HTTPS server.
# Might be either absolute path or path related to Tendermint's config directory.
# NOTE: both tls-cert-file and tls-key-file must be present for Tendermint to create HTTPS server.
# Otherwise, HTTP server is run.
tls_key_file = ""

# pprof listen address (https://golang.org/pkg/net/http/pprof)
pprof_laddr = "localhost:6060"

#######################################################
###           P2P Configuration Options             ###
#######################################################
[p2p]

# Address to listen for incoming connections
laddr = "tcp://0.0.0.0:26656"

# Address to advertise to peers for them to dial
# If empty, will use the same port as the laddr,
# and will introspect on the listener or use UPnP
# to figure out the address. ip and port are required
# example: 159.89.10.97:26656
external_address = "45.33.55.23:26656"

# Comma separated list of seed nodes to connect to
seeds = ""

# Comma separated list of nodes to keep persistent connections to
#persistent_peers = "209688f5bccb88f6397a97cc11ab545a014aa559@137.184.92.115:26656,d45e9dd8878d7c22d59ded3557f61da37420a4c6@95.217.118.211:26656,cae7d9d21c1752300277eab72d861b0c6638b2e3@164.68.119.151:26656,7a44ea6ecb59b0e4bd01b58a75163ec64b164bb4@63.210.148.24:26656,3bf3d98dfd4000dd5ff8189882a9f96848b99b87@137.220.60.196:26656,057fa262fe2030cc6e9095dc52d15b79ffcb923d@142.115.20.25:26656"
persistent_peers="209688f5bccb88f6397a97cc11ab545a014aa559@137.184.92.115:26656,d45e9dd8878d7c22d59ded3557f61da37420a4c6@95.217.118.211:26656,cae7d9d21c1752300277eab72d861b0c6638b2e3@164.68.119.151:26656"

# UPNP port forwarding
upnp = false

# Path to address book
addr_book_file = "config/addrbook.json"

# Set true for strict address routability rules
# Set false for private or local networks
addr_book_strict = true

# Maximum number of inbound peers
max_num_inbound_peers = 333

# Maximum number of outbound peers to connect to, excluding persistent peers
max_num_outbound_peers = 333

# List of node IDs, to which a connection will be (re)established ignoring any existing limits
unconditional_peer_ids = ""

# Maximum pause when redialing a persistent peer (if zero, exponential backoff is used)
persistent_peers_max_dial_period = "0s"

# Time to wait before flushing messages out on the connection
flush_throttle_timeout = "100ms"

# Maximum size of a message packet payload, in bytes
max_packet_msg_payload_size = 1024

# Rate at which packets can be sent, in bytes/second
send_rate = 5120000

# Rate at which packets can be received, in bytes/second
recv_rate = 5120000

# Set true to enable the peer-exchange reactor
pex = true

# Seed mode, in which node constantly crawls the network and looks for
# peers. If another node asks it for addresses, it responds and disconnects.
#
# Does not work if the peer-exchange reactor is disabled.
seed_mode = false

# Comma separated list of peer IDs to keep private (will not be gossiped to other peers)
private_peer_ids = ""

# Toggle to disable guard against peers connecting from the same ip.
allow_duplicate_ip = false

# Peer connection configuration.
handshake_timeout = "30s"
dial_timeout = "5s"

#######################################################
###          Mempool Configuration Option          ###
#######################################################
[mempool]

recheck = true
broadcast = true
wal_dir = ""

# Maximum number of transactions in the mempool
size = 5000

# Limit the total size of all txs in the mempool.
# This only accounts for raw transactions (e.g. given 1MB transactions and
# max_txs_bytes=5MB, mempool will only accept 5 transactions).
max_txs_bytes = 1073741824

# Size of the cache (used to filter transactions we saw earlier) in transactions
cache_size = 12000

# Do not remove invalid transactions from the cache (default: false)
# Set to true if it's not possible for any invalid transaction to become valid
# again in the future.
keep-invalid-txs-in-cache = false

# Maximum size of a single transaction.
# NOTE: the max size of a tx transmitted over the network is {max_tx_bytes}.
max_tx_bytes = 1048576

# Maximum size of a batch of transactions to send to a peer
# Including space needed by encoding (one varint per transaction).
# XXX: Unused due to https://github.com/tendermint/tendermint/issues/5796
max_batch_bytes = 0

#######################################################
###         State Sync Configuration Options        ###
#######################################################
[statesync]
# State sync rapidly bootstraps a new node by discovering, fetching, and restoring a state machine
# snapshot from peers instead of fetching and replaying historical blocks. Requires some peers in
# the network to take and serve state machine snapshots. State sync is not attempted if the node
# has any local state (LastBlockHeight > 0). The node will have a truncated block history,
# starting from the height of the snapshot.
enable = false

# RPC servers (comma-separated) for light client verification of the synced state machine and
# retrieval of state data for node bootstrapping. Also needs a trusted height and corresponding
# header hash obtained from a trusted source, and a period during which validators can be trusted.
#
# For Cosmos SDK-based chains, trust_period should usually be about 2/3 of the unbonding time (~2
# weeks) during which they can be financially punished (slashed) for misbehavior.
rpc_servers = ""
trust_height = 0
trust_hash = ""
trust_period = "168h0m0s"

# Time to spend discovering snapshots before initiating a restore.
discovery_time = "15s"

# Temporary directory for state sync snapshot chunks, defaults to the OS tempdir (typically /tmp).
# Will create a new, randomly named directory within, and remove it when done.
temp_dir = ""

# The timeout duration before re-requesting a chunk, possibly from a different
# peer (default: 1 minute).
chunk_request_timeout = "10s"

# The number of concurrent chunk fetchers to run (default: 1).
chunk_fetchers = "4"

#######################################################
###       Fast Sync Configuration Connections       ###
#######################################################
[fastsync]

# Fast Sync version to use:
#   1) "v0" (default) - the legacy fast sync implementation
#   2) "v1" - refactor of v0 version for better testability
#   2) "v2" - complete redesign of v0, optimized for testability & readability
version = "v0"

#######################################################
###         Consensus Configuration Options         ###
#######################################################
[consensus]

wal_file = "data/cs.wal/wal"

# How long we wait for a proposal block before prevoting nil
timeout_propose = "3s"
# How much timeout_propose increases with each round
timeout_propose_delta = "500ms"
# How long we wait after receiving +2/3 prevotes for “anything” (ie. not a single block or nil)
timeout_prevote = "1s"
# How much the timeout_prevote increases with each round
timeout_prevote_delta = "500ms"
# How long we wait after receiving +2/3 precommits for “anything” (ie. not a single block or nil)
timeout_precommit = "1s"
# How much the timeout_precommit increases with each round
timeout_precommit_delta = "500ms"
# How long we wait after committing a block, before starting on the new
# height (this gives us a chance to receive some more precommits, even
# though we already have +2/3).
timeout_commit = "1s"

# How many blocks to look back to check existence of the node's consensus votes before joining consensus
# When non-zero, the node will panic upon restart
# if the same consensus key was used to sign {double_sign_check_height} last blocks.
# So, validators should stop the state machine, wait for some blocks, and then restart the state machine to avoid panic.
double_sign_check_height = 0

# Make progress as soon as we have all the precommits (as if TimeoutCommit = 0)
skip_timeout_commit = false

# EmptyBlocks mode and possible interval between empty blocks
create_empty_blocks = true
create_empty_blocks_interval = "0s"

# Reactor sleep duration parameters
peer_gossip_sleep_duration = "100ms"
peer_query_maj23_sleep_duration = "2s"

#######################################################
###   Transaction Indexer Configuration Options     ###
#######################################################
[tx_index]

# What indexer to use for transactions
#
# The application will set which txs to index. In some cases a node operator will be able
# to decide which txs to index based on configuration set in the application.
#
# Options:
#   1) "null"
#   2) "kv" (default) - the simplest possible indexer, backed by key-value storage (defaults to levelDB; see DBBackend).
#               - When "kv" is chosen "tx.height" and "tx.hash" will always be indexed.
indexer = "kv"

#######################################################
###       Instrumentation Configuration Options     ###
#######################################################
[instrumentation]

# When true, Prometheus metrics are served under /metrics on
# PrometheusListenAddr.
# Check out the documentation for the list of available metrics.
prometheus = false

# Address to listen for Prometheus collector(s) connections
prometheus_listen_addr = ":26660"

# Maximum number of simultaneous connections.
# If you want to accept a larger number than the default, make sure
# you increase your OS limits.
# 0 - unlimited.
max_open_connections = 3

# Instrumentation namespace
namespace = "tendermint"
```

**genesis.toml**
```
 cat config/genesis.json
{
  "genesis_time": "2021-12-17T04:47:16.158256404Z",
  "chain_id": "vidulum-1",
  "initial_height": "1",
  "consensus_params": {
    "block": {
      "max_bytes": "22020096",
      "max_gas": "-1",
      "time_iota_ms": "1000"
    },
    "evidence": {
      "max_age_num_blocks": "100000",
      "max_age_duration": "172800000000000",
      "max_bytes": "1048576"
    },
    "validator": {
      "pub_key_types": [
        "ed25519"
      ]
    },
    "version": {}
  },
  "app_hash": "",
  "app_state": {
    "auth": {
      "params": {
        "max_memo_characters": "256",
        "tx_sig_limit": "7",
        "tx_size_cost_per_byte": "10",
        "sig_verify_cost_ed25519": "590",
        "sig_verify_cost_secp256k1": "1000"
      },
      "accounts": []
    },
    "bank": {
      "params": {
        "send_enabled": [],
        "default_send_enabled": true
      },
      "balances": [],
      "supply": [],
      "denom_metadata": []
    },
    "capability": {
      "index": "1",
      "owners": []
    },
    "crisis": {
      "constant_fee": {
        "denom": "stake",
        "amount": "1000"
      }
    },
    "distribution": {
      "params": {
        "community_tax": "0.020000000000000000",
        "base_proposer_reward": "0.010000000000000000",
        "bonus_proposer_reward": "0.040000000000000000",
        "withdraw_addr_enabled": true
      },
      "fee_pool": {
        "community_pool": []
      },
      "delegator_withdraw_infos": [],
      "previous_proposer": "",
      "outstanding_rewards": [],
      "validator_accumulated_commissions": [],
      "validator_historical_rewards": [],
      "validator_current_rewards": [],
      "delegator_starting_infos": [],
      "validator_slash_events": []
    },
    "evidence": {
      "evidence": []
    },
    "feegrant": {
      "allowances": []
    },
    "genutil": {
      "gen_txs": []
    },
    "gov": {
      "starting_proposal_id": "1",
      "deposits": [],
      "votes": [],
      "proposals": [],
      "deposit_params": {
        "min_deposit": [
          {
            "denom": "stake",
            "amount": "10000000"
          }
        ],
        "max_deposit_period": "172800s"
      },
      "voting_params": {
        "voting_period": "172800s"
      },
      "tally_params": {
        "quorum": "0.334000000000000000",
        "threshold": "0.500000000000000000",
        "veto_threshold": "0.334000000000000000"
      }
    },
    "ibc": {
      "client_genesis": {
        "clients": [],
        "clients_consensus": [],
        "clients_metadata": [],
        "params": {
          "allowed_clients": [
            "06-solomachine",
            "07-tendermint"
          ]
        },
        "create_localhost": false,
        "next_client_sequence": "0"
      },
      "connection_genesis": {
        "connections": [],
        "client_connection_paths": [],
        "next_connection_sequence": "0",
        "params": {
          "max_expected_time_per_block": "30000000000"
        }
      },
      "channel_genesis": {
        "channels": [],
        "acknowledgements": [],
        "commitments": [],
        "receipts": [],
        "send_sequences": [],
        "recv_sequences": [],
        "ack_sequences": [],
        "next_channel_sequence": "0"
      }
    },
    "mint": {
      "minter": {
        "inflation": "0.130000000000000000",
        "annual_provisions": "0.000000000000000000"
      },
      "params": {
        "mint_denom": "stake",
        "inflation_rate_change": "0.130000000000000000",
        "inflation_max": "0.200000000000000000",
        "inflation_min": "0.070000000000000000",
        "goal_bonded": "0.670000000000000000",
        "blocks_per_year": "6311520"
      }
    },
    "params": null,
    "slashing": {
      "params": {
        "signed_blocks_window": "100",
        "min_signed_per_window": "0.500000000000000000",
        "downtime_jail_duration": "600s",
        "slash_fraction_double_sign": "0.050000000000000000",
        "slash_fraction_downtime": "0.010000000000000000"
      },
      "signing_infos": [],
      "missed_blocks": []
    },
    "staking": {
      "params": {
        "unbonding_time": "1814400s",
        "max_validators": 100,
        "max_entries": 7,
        "historical_entries": 10000,
        "bond_denom": "stake"
      },
      "last_total_power": "0",
      "last_validator_powers": [],
      "validators": [],
      "delegations": [],
      "unbonding_delegations": [],
      "redelegations": [],
      "exported": false
    },
    "transfer": {
      "port_id": "transfer",
      "denom_traces": [],
      "params": {
        "send_enabled": true,
        "receive_enabled": true
      }
    },
    "upgrade": {},
    "vesting": {},
    "vidulum": {}
  }
}
```

**Comment by hdmiimdh** [comment]
This looks harmful to me by the way @erikgrinaker 

Here's the use case I'm running into and notice it quite frequently, when I'm trying to state sync from RPC nodes and adding those nodes into peers the snapshot can never been discovered I always keep getting `Discovering snapshots for 15s module=statesync` followed by `err="auth failure: secret conn failed: read tcp..."`, so it prevents me from syncing completely. Is there anything that we can do, it looks very annoying?

**Comment by joshcaskiefoundry** [comment]
@hdmiimdh I have the same issue currently. Find any solution?
